### DISCUSSION TITLE
support gitlab iterations

### DISCUSSION THREAD

**Discussion by stuart-warren** [magit/forge#487] [Ideas]
Would it be possible to support pulling gitlab issue iterations into the database?

https://docs.gitlab.com/ee/user/group/iterations/ (essentially sprint)

I assume even if you didn't want to add built in support for `forge-list-iteration-issues` if it's not a cross forge feature, I assume I could add something myself for that?

**Comment by tarsius** [comment]
> I assume even if you didn't want to add built in support for `forge-list-iteration-issues` if it's not a cross forge feature,

Not only that, but it seems this additionally is not available in the foss variants of gitlab and is about to be deprecated. (I am unsure whether the replacement is also going to be limited to the non-free offerings. I assume it will be, for money making reasons. Also I have to admit I don't fully understand this feature, but I think I understand enough to know that I am not particularly interested.)

I have no intention of supporting this, not now, most likely ever. There are still many useful features left to support on Forge's side that are supported across more forges, are available in the foss-variants of gitlab (and other foss forges, as well as github, which while never foss, is still the most popular offering), and that also seem more useful to more users.

> I assume I could add something myself for that?

So yes, that. It probably won't be trivial though. You would likely have to add your own tables and classes, and there is little documentation and I won't have time to help you sort things out. In theory it should be possible to do without any changes to Forge or `closql`, but if it turns out that really isn't possible, then I *might* be willing to merge some *non-intrusive* patches.

Sorry for not being able to be more encouraging in this case.
